cypress/integration/brewery/ScenarioSharing.js

```diff
@@ -1,40 +1,211 @@
+    RolesEdition.getRoleEditorAgentName(USER_EXAMPLE.email).should('have.text', USER_EXAMPLE.email);
+    RolesEdition.getSelectWithAction(USER_EXAMPLE.email).first().click();
+    RolesEdition.getSelectActionName().click();
+    RolesEdition.getNoAdminErrorMessage().should('be.visible');
+    RolesEdition.getShareScenarioDialogSubmitButton().should('be.disabled');
   });
 
   it('can add & remove people, and change roles in scenario sharing pop-up', () => {
     // TODO: open share pop-up, make some changes (add/remove/change role + workspace default role), confirm & refresh
     // page to check changes have been made
     // For local run with stubbing, page refresh could be replaced by switching to another scenario and then switching
     // back
+    const scenario = UNSHARED_SCENARIOS_LIST[0];
+
+    const grantedPermissions = [PERMISSIONS.READ, PERMISSIONS.READ_SECURITY];
+    const notGrantedPermissions = [
+      PERMISSIONS.LAUNCH,
+      PERMISSIONS.WRITE,
+      PERMISSIONS.VALIDATE,
+      PERMISSIONS.WRITE_SECURITY,
+      PERMISSIONS.DELETE,
+    ];
+
+    Scenarios.selectScenario(scenario.name, scenario.id);
+    RolesEdition.getShareScenarioButton().click();
+    RolesEdition.getShareScenarioDialogTitle().should('have.text', 'Share ' + scenario.name);
+    RolesEdition.getSelectWithAction('Workspace').click();
+    RolesEdition.getSelectOption(ROLES.SCENARIO.EDITOR).click();
+    RolesEdition.addAgent(USERS_LIST[1].email);
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.EDITOR).should('be.checked');
+    RolesEdition.getShareScenarioDialogSecondCancelButton().click();
+    RolesEdition.getSelectWithAction('Workspace').click();
+    RolesEdition.getSelectOption(ROLES.SCENARIO.VALIDATOR).click();
+    RolesEdition.addAgent(USERS_LIST[1].email);
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.VALIDATOR).should('be.checked');
+    RolesEdition.getShareScenarioDialogSecondCancelButton().click();
+    RolesEdition.getSelectWithAction('Workspace').click();
+    RolesEdition.getSelectOption(ROLES.SCENARIO.ADMIN).click();
+    RolesEdition.addAgent(USERS_LIST[1].email);
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.ADMIN).should('be.checked');
+    RolesEdition.getShareScenarioDialogSecondCancelButton().click();
+    RolesEdition.getSelectWithAction('Workspace').click();
+    RolesEdition.getSelectOption(ROLES.SCENARIO.VIEWER).click();
+
+    RolesEdition.addAgent(USERS_LIST[1].email);
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.VIEWER).should('be.checked');
+
+    RolesEdition.checkShareScenarioDialogGrantedPermissionChip(grantedPermissions, notGrantedPermissions);
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.EDITOR).click();
+    RolesEdition.getShareScenarioDialogRolesCheckbox(ROLES.SCENARIO.VIEWER).should('not.be.checked');
+    for (let i = 0; i < 2; ++i) grantedPermissions.push(notGrantedPermissions.shift());
```

Comment:
Could we declare a dict with expected permissions for each role instead of writing loops & shift in the test ? It will be easier to read and maintain